libs/sdk-ui-filters/src/AttributeFilter/AttributeFilter.tsx

```diff
@@ -118,104 +134,86 @@ export interface IAttributeFilterProps {
     FilterError?: React.ComponentType<{ error?: any }>;
 }
 
-interface IAttributeFilterState {
-    title: string;
-    isLoading: boolean;
-    error?: any;
-}
-
 const DefaultFilterError: React.FC = injectIntl(({ intl }) => {
     const text = intl.formatMessage({ id: "gs.filter.error" });
     return <div className="gd-message error s-button-error">{text}</div>;
 });
 
-class AttributeFilterCore extends React.PureComponent<IAttributeFilterProps, IAttributeFilterState> {
-    public static defaultProps = {
-        locale: "en-US",
-        FilterError: DefaultFilterError,
-        fullscreenOnMobile: false,
-        onError: defaultErrorHandler,
-        titleWithSelection: false,
+const AttributeFilterCore: React.FC<IAttributeFilterProps> = (props) => {
+    const {
+        locale,
+        workspace,
+        backend,
+        FilterError,
+        title,
+        titleWithSelection,
+        fullscreenOnMobile,
+        identifier,
+        filter,
+        parentFilters,
+        connectToPlaceholder,
+        parentFilterOverAttribute,
+        onError,
+        onApply,
+    } = props;
+
+    const resolvedParentFilters = useResolveValueWithPlaceholders(parentFilters);
+    const [placeholder, setPlaceholder] = usePlaceholder(connectToPlaceholder);
+
+    const currentFilter = placeholder || filter;
+
+    const getBackend = () => {
+        return backend.withTelemetry("AttributeFilter", props);
     };
 
-    public state: IAttributeFilterState = {
-        title: "",
-        error: null,
-        isLoading: false,
-    };
+    // todo add dependencies on filter/placeholder??
+    const {
+        result: attributeTitle,
+        error: attributeError,
+        status: attributeStatus,
+    } = useCancelablePromise(
+        {
+            promise: async () => {
+                const attributes = getBackend().workspace(workspace).attributes();
+                const displayForm = await attributes.getAttributeDisplayForm(
+                    getObjRef(currentFilter, identifier),
+                );
+                const attribute = await attributes.getAttribute(displayForm.attribute);
+
+                return attribute.title;
+            },
+        },
+        [identifier, workspace],
+    );
+
+    const onFilterApply = (selectedItems: IAttributeElement[], isInverted: boolean) => {
+        const useUriElements =
+            currentFilter && isAttributeElementsByRef(filterAttributeElements(currentFilter));
```

Comment:
This will not work if `currentFilter` is a placeholder.

Reply:
The placeholder variable renamed to `resolvedPlaceholder` to minimise confusion about variable name and type.